### PR TITLE
Feature serhiiyanchuk forgot password email

### PR DESCRIPTION
**Pull request by SerhiiYanchuk**

### CODE REVIEW

File: frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  public resetPassword(): void {
+    this.httpClientService.getRequest<boolean>
+    (`/Users/Email/${this.emailForm.get('userEmail')?.value}`).pipe(
+      mergeMap(isEmailExist => {
+        if (isEmailExist) {
+          const dto: ForgotPasswordDto = 
+          { 
+            email: this.emailForm.get('userEmail')?.value, 
+            clientURI: 'http://localhost:4200/reset-password',
```
Comment (Yaroslaww-1) [inline frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.ts:41]: Please take this from src/environments

File: frontend/ATS/src/app/users/components/reset-password-box/reset-password-box.component.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    [placeholder]="isPasswordHide ? '••••••••' : 'Enter your password'" autocomplete="on">
+                <mat-icon class="reset-password-show-password" matSuffix (click)="isPasswordHide = !isPasswordHide">
+                    {{isPasswordHide ? 'visibility_off' : 'visibility'}}</mat-icon>
+            </mat-form-field>
+            <div *ngIf="loginRegistCommonComponent.checkIsControlValid(resetPasswordForm,'userPasswordConfirmation')">
+                <div class="logo-validation-message"
+                    *ngIf="resetPasswordForm.controls['userPasswordConfirmation'].errors?.required">
+                    Please enter your password secoond time.
+                </div>
+            </div>
```
Comment (Yaroslaww-1) [inline frontend/ATS/src/app/users/components/reset-password-box/reset-password-box.component.html:42]: I can see a lot of code duplication here. Can it be fixed? You can move password input ```<mat-form-field>...</mat-form-field>``` to separate component and pass ```formControlName``` to it.

File: frontend/ATS/src/app/users/guards/reset-password.guard.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  public canActivate(
+    route: ActivatedRouteSnapshot, 
+    state: RouterStateSnapshot)
+    : Observable<boolean> {
+
+    const email: string = route.queryParams['email'];
+    const token: string = route.queryParams['token'];
+    return this.httpClientService.getRequest<boolean>
+    (`/Auth/reset-password?email=${email}&token=${token}`).pipe(
```
Comment (Yaroslaww-1) [inline frontend/ATS/src/app/users/guards/reset-password.guard.ts:20]: You can create ```auth.service.ts``` based on ```http-client.service.ts```. It secures you on reset password route change.

File: backend/src/Application/Auth/Commands/ForgotPasswordCommand.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Application.Auth.Commands
+{
+    public class ForgotPasswordCommand : IRequest<Unit>
+    {
+        public string Email { get; }
```
Comment (Yaroslaww-1) [inline backend/src/Application/Auth/Commands/ForgotPasswordCommand.cs:20]: In backend/src/Application/Auth/Commands/ResetPasswordCommand.cs you have used Dto as a class parameter. For consistency please change either this class to use Dto as a Command param or change ResetPasswordCommand to map Dto params to Command params. Personally I'd stick ResetPasswordCommand's way

File: frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.scss [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    margin-bottom: 10px;
+    
+}
+
+.dialog-back-button {   
+    cursor: pointer;
+}
+
+.mat-dialog-title span {
+    font-family   : $primary-font;
```
Comment (anton-matsyshyn) [inline frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.scss:22]: We don't have such align in app, please, remove it

File: frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.scss [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+.mat-button {
+    width : 160px;
+    height: 45px;
+}
+
+.dialog-validation-message {
+    font-family: $primary-font;
+    font-weight: normal;
+    font-size  : 12px;
```
Comment (anton-matsyshyn) [inline frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.scss:50]: Same as above

File: frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  styleUrls: ['./forgot-password-dialog.component.scss'],
+})
+export class ForgotPasswordDialogComponent {
+
+  constructor(
+    private dialogRef: MatDialogRef<ForgotPasswordDialogComponent>,
+    public loginRegistCommonComponent: LoginRegistCommonComponent,
+    private notificationService: NotificationService,
+    private authService: AuthenticationService) {}
+
```
Comment (anton-matsyshyn) [inline frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.ts:24]: Duplicated empty line

File: frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+          { 
+            email: this.emailForm.get('userEmail')?.value, 
+            clientURI: environment.resetPasswordClientUri, 
+          };
+          return this.authService.sendPasswordResetRequest(forgotPasswordDto);      
+        }
+        this.notificationService.showErrorMessage('There is no user with such email address.');
+        return EMPTY;
+      }),
+    )
```
Comment (anton-matsyshyn) [inline frontend/ATS/src/app/users/components/forgot-password-dialog/forgot-password-dialog.component.ts:48]: Better to move `pipe` call to new line and align in with subscribe call

File: frontend/ATS/src/app/users/services/auth.service.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { HttpClientService } from 'src/app/shared/services/http-client.service';
+import { ForgotPasswordDto } from '../models/forgot-password-dto';
+import { ResetPasswordDto } from '../models/reset-password-dto';
+
+@Injectable({ providedIn: 'root' })
+export class AuthenticationService {
+
```
Comment (anton-matsyshyn) [inline frontend/ATS/src/app/users/services/auth.service.ts:9]: You can remove one empty line here

File: frontend/ATS/src/environments/environment.prod.ts [hunk 10]
```diff
@@ -1,4 +1,5 @@
 export const environment = {
   production: true,
   apiUrl: 'https://localhost:5001/api',
+  resetPasswordClientUri: 'http://localhost:4200/reset-password',
```
Comment (anton-matsyshyn) [inline frontend/ATS/src/environments/environment.prod.ts:4]: Use environment variables for this. Such approach won't work in staging environment

File: frontend/ATS/src/environments/environment.ts [hunk 11]
```diff
@@ -5,6 +5,7 @@
 export const environment = {
   production: false,
   apiUrl: 'https://localhost:5001/api',
+  resetPasswordClientUri: 'http://localhost:4200/reset-password',
```
Comment (anton-matsyshyn) [inline frontend/ATS/src/environments/environment.ts:8]: Same as above

**Pull request merged**